ee/src/main/java/org/jboss/as/ee/subsystem/ManagedThreadFactoryResourceDefinition.java

```diff
@@ -40,6 +42,12 @@
  */
 public class ManagedThreadFactoryResourceDefinition extends SimpleResourceDefinition {
 
+    /**
+     * the resource definition's dynamic runtime capability
+     */
+    public static final RuntimeCapability<Void> CAPABILITY = RuntimeCapability.Builder.of("org.wildfly.ee.concurrent.threadfactory", true, ManagedThreadFactoryImpl.class)
```

Comment:
Can we not use javax.enterprise.concurrent.ManagedThreadFactory for the service type? Generally, we should avoid exposing specific implementation classes via the capability contract.